### PR TITLE
e2e: bootstrap vault and provision Nomad with vault tokens

### PR DESCRIPTION
**Pull request by tgross**
Provisions vault with the policies described in the Nomad Vault integration
guide ([ref](https://www.nomadproject.io/docs/integrations/vault-integration)), and drops a configuration file for Nomad vault server configuration
with its token. The vault root token is exposed to the E2E runner so that
tests can write additional policies to vault.

Supports https://github.com/hashicorp/nomad/issues/8845

### CODE REVIEW

File: e2e/terraform/packer/linux/setup.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ set -e
 
 # Will be overwritten at test time with the version specified
 NOMADVERSION=0.9.1
-CONSULVERSION=1.7.3
-VAULTVERSION=1.1.1
+CONSULVERSION=1.8.3
+VAULTVERSION=1.5.4
```
Comment (tgross) [inline e2e/terraform/packer/linux/setup.sh:10]: I had to upgrade the vault version to something recent because the policy in the learn guide requires it.

**Comment by tgross** [comment]
cc @angrycub as a heads up that this effectively exercises the learn guide too, which is a bonus.

File: e2e/terraform/vault.tf [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    type        = "ssh"
+    user        = "ubuntu"
+    host        = aws_instance.server[count.index].public_ip
+    port        = 22
+    private_key = file("${path.root}/keys/${local.random_name}.pem")
+  }
+}
+
+
+#    2020-10-02T15:32:53.298Z [ERROR] nomad.vault: failed to validate self token/role: retry=30s error="1 error occurred:
```
Comment (drewbailey) [inline e2e/terraform/vault.tf:67]: is this left over from debugging?

File: e2e/terraform/config/full-cluster/vault/server/vault.hcl [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
-  cluster_address = "{{ GetPrivateIP }}:8201"
-  tls_disable     = 1
+# this autounseal key is created by Terraform in the E2E infrastructure repo
+# and should be used only for these tests
+seal "awskms" {
+  region     = "us-east-1"
+  kms_key_id = "74b7e226-c745-4ddd-9b7f-2371024ee37d"
 }
+
+# Vault doesn't have autodiscovery for retry_join on its
```
Comment (drewbailey) [inline e2e/terraform/config/full-cluster/vault/server/vault.hcl:13]:
```suggestion
# Vault 1.5.4 doesn't have autodiscovery for retry_join on its
```
I'm sure the storage will stay as consul for awhile, but maybe including the version number will motivate a future reader to see if vault is ready

File: e2e/terraform/config/dev-cluster/vault/server/vault.hcl [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-  cluster_address = "{{ GetPrivateIP }}:8201"
-  tls_disable     = 1
+# this autounseal key is created by Terraform in the E2E infrastructure repo
+# and should be used only for these tests
+seal "awskms" {
+  region     = "us-east-1"
+  kms_key_id = "74b7e226-c745-4ddd-9b7f-2371024ee37d"
 }
+
+# Vault doesn't have autodiscovery for retry_join on its
```
Comment (drewbailey) [inline e2e/terraform/config/dev-cluster/vault/server/vault.hcl:13]:
```suggestion
# Vault 1.5.4 doesn't have autodiscovery for retry_join on its
```

**Review by drewbailey — APPROVED** [review]
lgtm!

File: e2e/terraform/vault.tf (hunk 2 above)
Comment (tgross) [reply e2e/terraform/vault.tf:67]: 😊

File: e2e/terraform/config/full-cluster/vault/server/vault.hcl (hunk 3 above)
Comment (tgross) [reply e2e/terraform/config/full-cluster/vault/server/vault.hcl:13]: Good call

**Pull request merged**